### PR TITLE
Enterprise account sharing image EASI

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMISImageDataSource_catalog
--- PASS: TestAccIBMISImageDataSource_catalog (69.42s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     71.190s
```
```
--- PASS: TestAccIBMISInstance_catalog (208.08s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     209.659s
```
```
=== RUN   TestAccIBMISImagesDataSource_catalog
--- PASS: TestAccIBMISImagesDataSource_catalog (45.84s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     47.475s
```

### CODE REVIEW

**Comment by astha-jain** [comment]
Instance provision with catalog offering - How is instance supporting catalog offering?

File: ibm/service/vpc/resource_ibm_is_instance.go [hunk 1]
```diff
@@ -341,6 +345,28 @@ func ResourceIBMISInstance() *schema.Resource {
...
+				ForceNew:    true,
+				Description: "The catalog offering or offering version to use when provisioning this virtual server instance. If an offering is specified, the latest version of that offering will be used. The specified offering or offering version may be in a different account in the same enterprise, subject to IAM policies.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						isInstanceCatalogOfferingOfferingCrn: {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Identifies a catalog offering by a unique CRN property",
+						},
+						isInstanceCatalogOfferingVersionCrn: {
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_instance.go:362]: should these be mutually exclusive ?

File: ibm/service/vpc/resource_ibm_is_snapshot.go [hunk 2]
```diff
@@ -280,7 +280,7 @@ func resourceIBMISSnapshotCreate(d *schema.ResourceData, meta interface{}) error
 	options.SnapshotPrototype = snapshotprototypeoptions
 
 	log.Printf("[DEBUG] Snapshot create")
-
+	options.SnapshotPrototype = snapshotprototypeoptions
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_snapshot.go:283]: this line is added twice

File: ibm/service/vpc/resource_ibm_is_instance.go (hunk 1 above)
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_instance.go:362]: add conflicts with snapshot, image, template

File: website/docs/d/is_image.html.markdown [hunk 3]
```diff
@@ -47,6 +47,12 @@ Review the argument references that you can specify for your data source.
 In addition to all argument reference list, you can access the following attribute references after your data source is created.
 
 - `architecture` - (String) The architecture of the image.
+- `catalog_offering` - (List) The catalog offering for this image.
+  Nested scheme for **catalog_offering**:
+  - `managed` - (String) Indicates whether this image is managed as part of a catalog offering. If an image is managed, accounts in the same enterprise with access to that catalog can specify the image's catalog offering version CRN to provision virtual server instances using the image.
```
Comment (deepaksibm) [inline website/docs/d/is_image.html.markdown:52]: Boolean

File: website/docs/d/is_images.html.markdown [hunk 4]
```diff
@@ -49,6 +50,12 @@ You can access the following attribute references after your data source is crea
   Nested scheme for `images`:
   - `architecture` - (String) The architecture for this image.
   - `crn` - (String) The CRN for this image.
+  - `catalog_offering` - (List) The catalog offering for this image.
+    Nested scheme for **catalog_offering**:
+    - `managed` - (String) Indicates whether this image is managed as part of a catalog offering. If an image is managed, accounts in the same enterprise with access to that catalog can specify the image's catalog offering version CRN to provision virtual server instances using the image.
```
Comment (deepaksibm) [inline website/docs/d/is_images.html.markdown:55]: Boolean

File: website/docs/r/is_instance.html.markdown [hunk 5]
```diff
@@ -321,7 +345,14 @@ Review the argument references that you can specify for your resource.
...
     ~> **Note:**
-    `snapshot` conflicts with `image` id and `instance_template`
+    `snapshot` conflicts with `image` id, `instance_template` and `catalog_offering`
+- `catalog_offering` - (Optional, List) The [catalog](https://cloud.ibm.com/docs/account?topic=account-restrict-by-user&interface=ui) offering or offering version to use when provisioning this virtual server instance. If an offering is specified, the latest version of that offering will be used. The specified offering or offering version may be in a different account in the same [enterprise](https://cloud.ibm.com/docs/account?topic=account-what-is-enterprise), subject to IAM policies.
+  Nested scheme for `catalog_offering`:
+  - `offering_crn` - (Optional, String) The CRN for this catalog offering. Identifies a catalog offering by this unique property
+  - `version_crn` - (Optional, String) The CRN for this version of a catalog offering. Identifies a version of a catalog offering by this unique property
+ 
+    ~> **Note:**
+    `offering_crn` conflicts with `version_crn`, both are mutually exclusive. `catalog_offering` and `image` id are mutually exclusive.
```
Comment (deepaksibm) [inline website/docs/r/is_instance.html.markdown:355]: `catalog_offering`, `image` id, `instance_template` and `boot_volume.0.snapshot` are mutually exclusive.

**Pull request closed without merging**